### PR TITLE
Ran clang-tidy with modernize-use-bool-literals

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/244/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/258/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1939/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/240/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/369/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1830/

**Comment by atsci** [comment]
clang-analyzer build *failed*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/502/

File: iocore/aio/AIO.cc [hunk 1]
```diff
@@ -399,7 +399,7 @@ aio_queue_req(AIOCallbackInternal *op, int fromAPI = 0)
 static inline int
 cache_op(AIOCallbackInternal *op)
 {
-  bool read = (op->aiocb.aio_lio_opcode == LIO_READ) ? 1 : 0;
+  bool read = (op->aiocb.aio_lio_opcode == LIO_READ) ? true : false;
```
Comment (SolidWallOfCode) [inline iocore/aio/AIO.cc]: Dang, it could have done a better job here. E.g.
`bool read = op->aiocb.aio_lio_opcode == LIO_READ;`

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/295/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/309/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1990/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/291/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/421/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1882/

**Comment by atsci** [comment]
clang-analyzer build *failed*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/554/

**Pull request merged**